evm/src/cpu/kernel/asm/bignum/modmul.asm

```diff
@@ -0,0 +1,192 @@
+// Arithmetic on little-endian integers represented with 128-bit limbs.
+// All integers must be under a given length bound, and are padded with leading zeroes.
+
+// Stores a * b % m in output_loc, leaving a, b, and m unchanged.
+// a, b, and m must have the same length.
+// Both output_loc and scratch_1 must have size length.
+// Both scratch_2 and scratch_3 have size 2 * length and be initialized with zeroes.
+global modmul_bignum:
```

Comment:
Please describe the algorithm below in a bit more detail. It's a really long function, written in assembly, and it's difficult to tell what's going on.

Reply:
I know there's already some comments but they get a bit lost with all the `// stack:` annotations (especially we don't have syntax highlighting for EVM assembly). Maybe surround them with some `***`s for emphasis?